### ISSUE TITLE
Identify actors and their responsibilities in the ossc

### ISSUE THREAD

**Issue by nadgowdas** [OpenSecureSupplyChain/ossc#8]
This issue is opened to track:

-  List of actors in the OSSC 
-  Role and responsibilities of every actor
- Interaction of every actor with the system (capture them in a workflow)

**Comment by nadgowdas** [comment]
Few actors I can think of are:

**1. Developer:** 

Responsibilities: 
a. Writes code
b. Declares third-party OSS dependencies for their app
c. Reviews recommendations from OSSC system around vulnerable dependencies and updates them as suggested.
d. Writes enough test-cases to ensure functional verifications (responsible for runtime bugs)

**2. CISO:**

There is no "one size fits all" solution for SSC, so it should be configurable to be made available in all sizes and shapes. And these size can be determined by security engineer/CISO for resp. org user. 

Responsibilities:
a. Defines organization wide policies that are input to various stages in the OSSC pipeline. This includes (but not limited to): trusted package repositories, allowed/forbidden packages to use, all policies around vulnerability management (threshold, time to fix, glass-break option, etc.)
b. Defines/certifies list of checks and tools to be used in OSSC.
c. Reviews summarized reports (in some form :)) to assess security posture of all apps in their org.

**3. Security Engineer:**

Responsibilities:
a. Overall architect to define and create OSSC.
b. Creates pipeline following all best-practices and includes tasks to perform all required security functions (e.g. SBOM generate, vulnerability scan, build, sign, etc.)
c. Setup different tools and performs system configurations (e.g. setup rekor, chains, keylime, etc.)
d. Responsible for ensuring secure configuration of pipeline (all keys, certs, password are stored and accessed securely, etc.)

**4. System Admin:**

Responsibilities:
a. Responsible to provide base cloud runtime (openshift or any cloud k8s runtime)
b. Responsible to ensure uptime for runtime and updating it as required.
c. Monitoring the runtime.

**Comment by nadgowdas** [comment]
To begin with, we should start with "Developer" and "Security Engineer"

**Comment by lukehinds** [comment]
Would end user be applicable here?

Another interesting one might be 'auditor'

**Comment by nadgowdas** [comment]
right, we should add them both. Does the following responsibilities for them make  sense ? 

Auditor:

Responsibilities:
a. Retrieve required attestation records from ossc (e.g. for a given image, they  can request provenance records, attestations etc.)
b. Review/Validate software provenance/attestation. 


End-User:

End-user is the real user using the software. E.g. a developer might build an `hello-world` app-image, while end-user is the one who  is hosting and running that  app-image on his runtime. 

Responsibilities:

a. Responsible for performing necessary validation checks before hosting the app-image. E.g. running admission-controller to  ensure he is running signed image. 
b. Responsible for monitoring and updating when new versions of the app-image is available.

**Comment by lukehinds** [comment]
lgtm @nadgowdas
